controller_manager/test/test_load_controller.cpp

```diff
@@ -584,3 +584,24 @@ TEST_F(TestLoadController, switch_multiple_controllers)
     ASSERT_EQ(lifecycle_msgs::msg::State::PRIMARY_STATE_INACTIVE, controller_if2->get_state().id());
   }
 }
+
+TEST_F(TestLoadController, can_set_and_get_non_default_update_rate)
```

Comment:
Please also add test where update-calls of controllers are tested. There is already a test with example of using a counter to check how many times a controller got updated.

Reply:
@destogl I have added another two test cases, however, I have no idea how the one about inactive controllers should look like. I want to check if the internal update rate of the CM stays zero if we're trying to update an inactive controller, but the way I'm doing it resulted in a nasty segfault when I checked it in the past. (see my commit)

Also, the tests don't seem to work currently, but I have no idea if the problem is on my part, because I get a runtime error in an unmodified test case here :

[ RUN      ] TestControllerManager.controller_lifecycle
[INFO] [1631212774.610695476] [test_controller_manager]: Configuring controller 'test_controller_name'
unknown file: Failure
C++ exception with description "expected [integer] got [not set]" thrown in the test body.

Reply:
Ok, I will have to check those.